api/src/test/nfl-teams.test.ts

```diff
@@ -0,0 +1,101 @@
+
+      const result = await request(app.getServer()).get(`${nflTeamsRoute.path}`); 
+      
+      expect(result.status).toEqual(200);
+      expect(result.body.data.length).toEqual(32);
+      expect(result.body.data[0].id).toEqual(mockNFLTeams[0].id);
+      expect(result.body.data[0].team_name_short).toEqual(mockNFLTeams[0].team_name_short);
+      expect(result.body.data[0].team_name_long).toEqual(mockNFLTeams[0].team_name_long);
+      expect(result.body.data[0].team_city_short).toEqual(mockNFLTeams[0].team_city_short);
+      expect(result.body.data[0].team_city_long).toEqual(mockNFLTeams[0].team_city_long);
+      expect(result.body.data[0].team_logo).toEqual(mockNFLTeams[0].team_logo);
+      expect(result.body.data).toContainEqual(mockNFLTeams[5]);
+      expect(result.body.data).toContainEqual(mockNFLTeams[10]);
+      expect(result.body.data).toContainEqual(mockNFLTeams[15]);
+      expect(result.body.data).toContainEqual(mockNFLTeams[20]);
+      expect(result.body.data).toContainEqual(mockNFLTeams[25]);
+      expect(result.body.data).toContainEqual(mockNFLTeams[30]);
+      expect(result.body.data).toContainEqual(mockNFLTeams[31]);
+    });
+  });
+
+  describe('[GET] /nfl-teams:id', () => {
+    it('response findOne nfl-team', async () => {
+
+      const mockNFLTeam : NFLTeam = {
+        id: 18,
+        team_name_short: 'Vikings',
+        team_name_long: 'Minnesota Vikings',
+        team_city_short: 'MIN',
+        team_city_long: 'Minnesota',
+        team_logo: 'image.png'
+      } ;
+      
+      DB.NFLTeams.findByPk = jest.fn().mockResolvedValue(mockNFLTeam);
+      (Sequelize as any).authenticate = jest.fn(); 
+
+      const result = await request(app.getServer()).get(`${nflTeamsRoute.path}/${mockNFLTeam.id}`); 
+      
+      expect(result.status).toEqual(200);
+      expect(result.body.data.id).toEqual(mockNFLTeam.id);
+      expect(result.body.data.team_name_short).toEqual(mockNFLTeam.team_name_short);
+      expect(result.body.data.team_name_long).toEqual(mockNFLTeam.team_name_long);
+      expect(result.body.data.team_city_short).toEqual(mockNFLTeam.team_city_short);
+      expect(result.body.data.team_city_long).toEqual(mockNFLTeam.team_city_long);
+      expect(result.body.data.team_logo).toEqual(mockNFLTeam.team_logo);
+    });
+
+    it('response findOne nfl-team - exception', async () => {
+
+      const mockNFLTeam : NFLTeam = {
+        id: 18,
+        team_name_short: 'Vikings',
+        team_name_long: 'Minnesota Vikings',
+        team_city_short: 'MIN',
+        team_city_long: 'Minnesota',
+        team_logo: 'image.png'
+      };
+      
+      DB.NFLTeams.findByPk = jest.fn().mockResolvedValue(undefined);
+      (Sequelize as any).authenticate = jest.fn(); 
+
+      const result = await request(app.getServer()).get(`${nflTeamsRoute.path}/${mockNFLTeam.id}`); 
+      
+      expect(result.status).toEqual(409);
+      expect(result.body.message).toEqual("NFL Team doesn't exist");
+    });
+
+  });
+
```

Comment:
You could maybe add one more test of checking for an exception again but this time pass in an ID but have that ID be out of bounds so like 44.

Reply:
oh that's a good idea, will do

Reply:
Oh yeah I did not even think of that way good catch.